src/sorta.nim

```diff
@@ -409,45 +414,79 @@ proc `$`*[Key, Val](b: SortedTable[Key, Val]): string =
 
 proc len*[Key, Val](b: SortedTable[Key, Val]): int {.inline.} = b.entries
 
-
-iterator entries[Key, Val](b: SortedTable[Key, Val]): Entry[Key, Val] =
-  template populateWithLeftmostChildren(stack, currentNode) =
-    while not currentNode.isNil:
-      for i in countdown(currentNode.m-1, 0):
-        stack.add(currentNode.e[i])
-      currentNode = currentNode.p0
-
-  var
-    root = b.root
-    currentEntry: Entry[Key, Val]
-    stack = newSeq[Entry[Key, Val]]()
-
-  stack.populateWithLeftmostChildren(root)
-  while stack.len > 0:
-    currentEntry = stack.pop()
-    yield currentEntry
-    stack.populateWithLeftmostChildren(currentEntry.p)
-
-
-iterator mentries[Key, Val](b: var SortedTable[Key, Val]): var Entry[Key, Val] =
-  template populateWithLeftmostChildren(stack, cntNode) =
-    var currentNode = cntNode
-    while not currentNode[].isNil:
-      for i in countdown(currentNode[].m-1, 0):
-        stack.add(addr currentNode[].e[i])
-      currentNode = addr currentNode[].p0
-
-  var
-    root = unsafeAddr b.root
-    currentEntry: ptr Entry[Key, Val]
-    stack = newSeq[ptr Entry[Key, Val]]()
-
-  stack.populateWithLeftmostChildren(root)
-  while stack.len > 0:
-    currentEntry = stack.pop()
-    yield currentEntry[]
-    stack.populateWithLeftmostChildren(addr currentEntry[].p)
-
+proc key[Key, Val](cursor: CursorPosition[Key, Val]): Key =
```

Comment:
Using the same variable name `cursor` for both `CursorPosition[Key, Val]` and for `Cursor[Key, Val]` is a bit misleading/confusing, can we change one of those?
Having in mind the comment above, maybe the second one can be `stack`, `curStack`, or something like that?

Reply:
I would like to keep "cursor" for Cursor, and I renamed CursorPosition variables as "position". Ok?